File: Classes/code/objRobot/OGLRen/Renderer.js
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a college course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Computer Graphics 
//           Rensselaer Polytechnic Institute
//  Date:    Sept 2015
//
//////////////////////////////////////////////////////////////////////////

var OGLRen = OGLRen || {};

/**
  * WebGL Light class
  */
OGLRen.Renderer = function(lnum) {
    
    /** some definitions for specifying the drawing mode */
	var POLYGON = 1, 
		LINE = 2, 
		TRIANGLE = 3, 
		LINE_LOOP = 4, 
		POINT = 5, 
		POLYLINE = 6;

}
OGLRen.Renderer.inheritsFrom(OGLRen.Light);

/** Add an Actor to the scene */
OGLRen.Renderer.prototype.addActor = function( actor) {};

/** Add an Light to the scene */
OGLRen.Renderer.prototype.addLight = function( light) {};
/** Turn lighting off if there are no normals */
OGLRen.Renderer.prototype.lighting = function( onOrOff) {};

/** Add an Camera to the scene */
OGLRen.Renderer.prototype.addCamera = function( cam) {};

/** Get the current active Camera, currently only one camera
  * supported
  */
OGLRen.Renderer.prototype.getCamera = function() {};

/** Sets the render mode for consequent calls */
OGLRen.Renderer.prototype.beginDraw = function( renderMode) {};

/** Signal all vertices have been entered, flush the buffer */
OGLRen.Renderer.prototype.endDraw = function() {};

/** specify a vector for the previous begin command */
OGLRen.Renderer.prototype.vertex = function( p) {};
/** specify a texture coord for the vertex */
OGLRen.Renderer.prototype.texCoord = function( p) {};
/** specify a normal for the vertex */
OGLRen.Renderer.prototype.normal = function( n) {};

/** Lets draw something, calls all objects render method,
  * Camera first, then Lights and Actors. Then swap buffers
  */
OGLRen.Renderer.prototype.render = function() {};

/** Lets make sure OpenGL is initialized with some safe settings */
OGLRen.Renderer.prototype.initialize = function() {};

/** Set the current rendering color */
OGLRen.Renderer.prototype.setMaterial = function( mat) {};

/** Push the current Model transformation matrix */
OGLRen.Renderer.prototype.pushModelMatrix = function() {};

/** Set the current orientation for consequent drawing */
OGLRen.Renderer.prototype.setOrientation = function( rotations,  scale, pos) {};

/** Pop the current Model transformation matrix */
OGLRen.Renderer.prototype.popModelMatrix = function() {};

/** Load the current texture
 */
OGLRen.Renderer.prototype.loadTexture = function( ImageNum,  atex) {};

/** Get/Set the width and Height of the render window */
OGLRen.Renderer.prototype.getWidth = function() {};
OGLRen.Renderer.prototype.getHeight = function() {};
OGLRen.Renderer.prototype.setSize = function( w,  h) {};

File: ClassPrevious/class/99/Hw3/zhaos/Renderer.java
//////////////////////////////////////////////////////////////////////////
//
//  This code is for instructional purposes only. It was generated for
//  use in a graduate level course to show certain aspects of data
//  storage algorithms. It has problems and should not be used
//  outside the class environment.
//
//  Author:  Thomas D. Citriniti     citrit@rpi.edu
//  Class:   Advanced Computer Graphics and Data visualization
//           Rensselaer Polytechnic Institute
//  Date:    January 1998
//
/////////////////////////////////////////////////////////////////////////

/**
 * Renderer interface, this is a place holder for the function calls 
 * in the Renderer class. Defined as an interface to specify behavior
 * not actions.
 *
 * @author		Thomas D. Citrinit
 * @version		0.1
 */
public interface Renderer
{
	/** some definitions for specifying the drawing mode */
	final int POLYGON =1, 
		LINE = 2, 
		TRIANGLE = 3, 
		LINE_LOOP = 4, 
		POINT = 5, 
		POLYLINE = 6;

	/** Add an Actor to the scene */
    public abstract void addActor(Actor actor);

	/** Add an Light to the scene */
    public abstract void addLight(Light light);

	/** Add an Camera to the scene */
    public abstract void addCamera(Camera cam);

	/** Get the current active Camera, currently only one camera
	  * supported
	  */
    public abstract Camera getCamera();

	/** Sets the render mode for consequent calls */
    public abstract void beginDraw(int renderMode);

	/** Signal all vertices have been entered, flush the buffer */
    public abstract void endDraw();

	/** specify a vector for the previous begin command */
    public abstract void vertex(PointType p);

	/** Lets draw something, calls all objects render method,
	  * Camera first, then Lights and Actors. Then swap buffers
	  */
    public abstract void render();
	
	/* Display list functions */
    public abstract void beginDL(int DL);
    public abstract void endDL();
    public abstract void callDL(int DL);
    public abstract int generateDL();

	/** Lets make sure OpenGL is initialized with some safe settings */
    public abstract void initialize();

	/** Set the current rendering color */
    public abstract void setMaterial(Material mat);

	/** Push the current Model transformation matrix */
	public abstract void pushModelMatrix();

	/** Set the current orientation for consequent drawing */
	public abstract void setOrientation(float rotations[], float scale[], 
										PointType pos);

	/** Pop the current Model transformation matrix */
	public abstract void popModelMatrix();

	/** Get/Set the width and Height of the render window */
	public abstract int getWidth();
	public abstract int getHeight();
	public abstract void setSize(int w, int h);


}
